src/Entity/Payment.php

```diff
@@ -2,192 +2,148 @@
+use Symfony\Component\Validator\Constraints as Assert;
 
 #[ApiResource(
     operations: [
         new Get(),
         new GetCollection(),
-        new Post(),
-        new Put(),
-        new Delete(),
-    ]
+        new Post(
+            denormalizationContext: ['groups' => 'payment:create'],
+        ),
+        new Put(
+            denormalizationContext: ['groups' => 'payment:update']
+        ),
+        new Delete()
+    ],
+    formats: ['json'],
+    denormalizationContext: [
+        'groups' => [
+            'payment:create',
+            'payment:update'
+        ]
+    ],
+    collectDenormalizationErrors: true
 )]
+#[ApiFilter(filterClass: SearchFilter::class, properties: ['event', 'member', 'name' => 'partial'])]
+#[ApiFilter(filterClass: RangeFilter::class, properties: ['date'])]
+#[ApiFilter(filterClass: OrderFilter::class, properties: ['date' => 'ASC', 'name' => 'ASC'])]
 #[ORM\Entity(repositoryClass: PaymentRepository::class)]
 class Payment
 {
-
     #[ORM\Id]
     #[ORM\Column(length: 255, unique: true)]
+    #[Assert\NotBlank]
+    #[Groups(['payment:create'])]
     private string $uid;
 
-    #[ORM\Column(length: 255)]
-    private string $name;
-
-    #[ORM\Column]
-    private int $amount;
-
-    #[ORM\Column(type: Types::DATETIME_MUTABLE)]
-    private ?\DateTimeInterface $date = null;
-
     #[ORM\ManyToOne]
-    #[ORM\JoinColumn(name: 'event_uid', referencedColumnName: 'uid', nullable: false)]
-    private PartyEvent $eventUid;
+    #[ORM\JoinColumn(name: 'event_uid', referencedColumnName: 'uid', nullable: false, onDelete: "CASCADE")]
+    #[Assert\NotBlank]
+    #[Groups(['payment:create'])]
+    private PartyEvent $event;
 
     #[ORM\ManyToOne]
-    #[ORM\JoinColumn(name: 'user_uid', referencedColumnName: 'uid', nullable: false)]
-    private User $userUid;
-
-    #[ORM\OneToMany(mappedBy: 'paymentUid', targetEntity: CalculationModification::class, orphanRemoval: true)]
-    private Collection $calculationModifications;
+    #[ORM\JoinColumn(name: 'member_uid', referencedColumnName: 'uid', nullable: false, onDelete: "CASCADE")]
+    #[Groups(['payment:create'])]
+    #[EventMember]
+    private Member $member;
 
-    #[ORM\OneToMany(mappedBy: 'paymentUid', targetEntity: ExcludeModification::class, orphanRemoval: true)]
-    private Collection $excludeModifications;
+    #[ORM\Column(length: 255)]
+    #[Assert\NotBlank]
+    #[Groups(['payment:create', 'payment:update'])]
+    private string $name;
 
-    public function __construct()
-    {
-        $this->calculationModifications = new ArrayCollection();
-        $this->excludeModifications = new ArrayCollection();
-    }
+    #[ORM\Column(type: Types::BIGINT)]
+    #[Assert\GreaterThan(0)]
+    #[Groups(['payment:create', 'payment:update'])]
+    private int $amount;
 
-    public function getUid(): string
-    {
-        return $this->uid;
-    }
+    #[ORM\Column(type: Types::BIGINT, nullable: true)]
+    #[Groups(['payment:create', 'payment:update'])]
```

Comment:
### ❗ Issue: Date can't be updated

I don't think we should have an option to update this field... but who knows... maybe it will be useful in future